### DISCUSSION TITLE
Ender 3 S1 F4 - UBL needed for AML 2.0?

### DISCUSSION THREAD

**Discussion by codegain** [mriscoc/Ender3V2S1#1203] [Q&A] [answered]
Hi, I'm currently running Ender 3 S1 F4 with UBL (Ender3S1-F4-UBL-20230904.bin) and I am curious to upgrade to the new version 20231202. But I can't figure out if I need UBL in order for AML 2.0 to function.

On the wiki page https://github.com/mriscoc/Ender3V2S1/wiki/Adaptive-Mesh-Leveling-(AML) it only says

> For be able to use AML it is necessary to have a **firmware with support for AML** and the proper plugin/post-processing script installed in the slicer.

I know where to get the processing scripts, but no firmware seems to have "AML" in their name which leads me to believe that all firmwares are capable of using AML 2.0 with the new version. Is that correct? Or which printers have support for AML currently?

**Comment by mriscoc** [comment, marked as answer]
All Anniversary 2023 firmwares are AML 2.0 capable.

**Reply by HankLumbard** [reply]
Can you tell me if this is compatible?

FIRMWARE_NAME:Marlin 2.1.3 MRiscoC Ender3V2-422-BLTUBL-LA-MPC, based on bugfix-2.1.x (Dec 15 2023 12:57:55) SOURCE_CODE_URL:github.com/mriscoc/Ender3V2S1 PROTOCOL_VERSION:1.0 MACHINE_TYPE:Ender3V2-422-BLTUBL-LA-MPC

**Reply by mriscoc** [reply]
Yes

**Reply by MaikoVoigt** [reply]
I also use an Ender S1 F4. Unfortunately, the 2023 Anniversary firmware is no longer available for download. Which firmware supports AML 2.0? Kind regards.

**Reply by mriscoc** [reply]
All firmwares since the Anniversary 2023 supports AML 2.0
